precompiles/referenda/src/lib.rs

```diff
@@ -356,18 +374,41 @@ where
 	/// * index: The index of a closed referendum whose Decision Deposit has not yet been refunded.
 	#[precompile::public("refundDecisionDeposit(uint32)")]
 	fn refund_decision_deposit(handle: &mut impl PrecompileHandle, index: u32) -> EvmResult {
-		let event = log1(
-			handle.context().address,
-			SELECTOR_LOG_DECISION_DEPOSIT_REFUNDED,
-			EvmDataWriter::new().write::<u32>(index).build(),
-		);
-		handle.record_log_costs(&[&event])?;
+		handle.record_log_costs_manual(1, 32 * 3)?;
+		// Get refunding deposit before dispatch
+		handle.record_cost(RuntimeHelper::<Runtime>::db_read_gas_cost())?;
+		let refunded_deposit: U256 = match ReferendumInfoFor::<Runtime>::get(index)
+			.ok_or(RevertReason::custom("Referendum index does not exist").in_field("index"))?
+		{
+			ReferendumInfo::Ongoing(x) => {
+				if let Some(d) = x.decision_deposit {
+					d.amount.into()
+				} else {
+					U256::zero()
+				}
+			}
```

Comment:
I think removing this block would be better than keeping it. 
```suggestion
```

Regardless, the refund_decision_deposit dispatch will fail if it is Ongoing...

Reply:
Totally, thank you